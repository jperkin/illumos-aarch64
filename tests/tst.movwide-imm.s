/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2017 Joyent, Inc.
 */

/*
 * C3.4.5 Move wide (immediate)
 */
.macro movw_imm op
	\op	w0, #0x0000
	\op	w0, #0x0000, lsl #0
	\op	w0, #0x0000, lsl #16
	\op	x0, #0x0000
	\op	x0, #0x0000, lsl #0
	\op	x0, #0x0000, lsl #16
	\op	x0, #0x0000, lsl #32
	\op	x0, #0x0000, lsl #48
	\op	w0, #0xffff
	\op	w0, #0xffff, lsl #0
	\op	w0, #0xffff, lsl #16
	\op	x0, #0xffff
	\op	x0, #0xffff, lsl #0
	\op	x0, #0xffff, lsl #16
	\op	x0, #0xffff, lsl #32
	\op	x0, #0xffff, lsl #48
.endm

.text
.globl libdis_test
.type libdis_test, @function
libdis_test:
	/*
	 * C5.6.127 MOVN
	 * C5.6.122 MOV (inverted wide immediate) (alias)
	 */
	movw_imm movn
	movw_imm movz
	movw_imm movk
.size libdis_test, [.-libdis_test]
