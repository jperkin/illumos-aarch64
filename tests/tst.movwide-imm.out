    libdis_test:      00 00 80 12  mov w0, #0xffffffff
    libdis_test+0x4:  00 00 80 12  mov w0, #0xffffffff
    libdis_test+0x8:  00 00 a0 12  movn w0, #0x0, lsl #16
    libdis_test+0xc:  00 00 80 92  mov x0, #0xffffffffffffffff
    libdis_test+0x10: 00 00 80 92  mov x0, #0xffffffffffffffff
    libdis_test+0x14: 00 00 a0 92  movn x0, #0x0, lsl #16
    libdis_test+0x18: 00 00 c0 92  movn x0, #0x0, lsl #32
    libdis_test+0x1c: 00 00 e0 92  movn x0, #0x0, lsl #48
    libdis_test+0x20: e0 ff 9f 12  movn w0, #0xffff
    libdis_test+0x24: e0 ff 9f 12  movn w0, #0xffff
    libdis_test+0x28: e0 ff bf 12  movn w0, #0xffff, lsl #16
    libdis_test+0x2c: e0 ff 9f 92  mov x0, #0xffffffffffff0000
    libdis_test+0x30: e0 ff 9f 92  mov x0, #0xffffffffffff0000
    libdis_test+0x34: e0 ff bf 92  mov x0, #0xffffffff0000ffff
    libdis_test+0x38: e0 ff df 92  mov x0, #0xffff0000ffffffff
    libdis_test+0x3c: e0 ff ff 92  mov x0, #0xffffffffffff
    libdis_test+0x40: 00 00 80 52  mov w0, #0x0
    libdis_test+0x44: 00 00 80 52  mov w0, #0x0
    libdis_test+0x48: 00 00 a0 52  movz w0, #0x0, lsl #16
    libdis_test+0x4c: 00 00 80 d2  mov x0, #0x0
    libdis_test+0x50: 00 00 80 d2  mov x0, #0x0
    libdis_test+0x54: 00 00 a0 d2  movz x0, #0x0, lsl #16
    libdis_test+0x58: 00 00 c0 d2  movz x0, #0x0, lsl #32
    libdis_test+0x5c: 00 00 e0 d2  movz x0, #0x0, lsl #48
    libdis_test+0x60: e0 ff 9f 52  mov w0, #0xffff
    libdis_test+0x64: e0 ff 9f 52  mov w0, #0xffff
    libdis_test+0x68: e0 ff bf 52  mov w0, #0xffff0000
    libdis_test+0x6c: e0 ff 9f d2  mov x0, #0xffff
    libdis_test+0x70: e0 ff 9f d2  mov x0, #0xffff
    libdis_test+0x74: e0 ff bf d2  mov x0, #0xffff0000
    libdis_test+0x78: e0 ff df d2  mov x0, #0xffff00000000
    libdis_test+0x7c: e0 ff ff d2  mov x0, #0xffff000000000000
    libdis_test+0x80: 00 00 80 72  movk w0, #0x0
    libdis_test+0x84: 00 00 80 72  movk w0, #0x0
    libdis_test+0x88: 00 00 a0 72  movk w0, #0x0, lsl #16
    libdis_test+0x8c: 00 00 80 f2  movk x0, #0x0
    libdis_test+0x90: 00 00 80 f2  movk x0, #0x0
    libdis_test+0x94: 00 00 a0 f2  movk x0, #0x0, lsl #16
    libdis_test+0x98: 00 00 c0 f2  movk x0, #0x0, lsl #32
    libdis_test+0x9c: 00 00 e0 f2  movk x0, #0x0, lsl #48
    libdis_test+0xa0: e0 ff 9f 72  movk w0, #0xffff
    libdis_test+0xa4: e0 ff 9f 72  movk w0, #0xffff
    libdis_test+0xa8: e0 ff bf 72  movk w0, #0xffff, lsl #16
    libdis_test+0xac: e0 ff 9f f2  movk x0, #0xffff
    libdis_test+0xb0: e0 ff 9f f2  movk x0, #0xffff
    libdis_test+0xb4: e0 ff bf f2  movk x0, #0xffff, lsl #16
    libdis_test+0xb8: e0 ff df f2  movk x0, #0xffff, lsl #32
    libdis_test+0xbc: e0 ff ff f2  movk x0, #0xffff, lsl #48
